File: sistema_java/codigos/oficinapoo/ItemAgenda.java
package oficinapoo;

public class ItemAgenda implements Comparable<ItemAgenda>
{
    public DataHora data;
    public Paciente paciente;

    private String descricao;

    //construtores
    public ItemAgenda()
    {
        descricao = " ";
        data = new DataHora();
        paciente = new Paciente();
    }
    
     public ItemAgenda(String descricao)
    {
       if (!setDescricao(descricao))
	{
            this.descricao = " ";
	}
        data = new DataHora();
        paciente = new Paciente();
    }
     
    public ItemAgenda(DataHora data, Paciente paciente, String descricao)
    {
        this.data = new DataHora(data);
        this.paciente = new Paciente(paciente);
        
        if (!setDescricao(descricao))
	{
            this.descricao = " ";
	}
    }
    
    public ItemAgenda(ItemAgenda itemAgenda)
    {
        this.data = new DataHora(itemAgenda.data);
        this.paciente = new Paciente(itemAgenda.paciente);
        this.descricao = itemAgenda.descricao;
    }
	
    //getters
    public String getDescricao()
    {
        return descricao;
    }

    //setters
    public boolean setDescricao(String descricao)
    {
        if (descricao.equals("") || descricao.equals(" "))//conferir se nome eh valido, validacao fraca mas ajuda um pouco
	{
            return false;
	}

	this.descricao = descricao;

	return true;
    }

    //sobrecarga do operador 'printf'
    public String toString()
    {
        return String.format("Data: %s Paciente: %s CPF: %s %s",this.data,
                              this.paciente.getNome(), this.paciente.getCpf(), this.descricao);
    }
    
    //metodo comparavel
    public int compareTo(ItemAgenda item)
    {
        return this.data.compareTo(item.data);
    }
	
}

File: sistema_cpluplus/codigos/ItemAgenda.cpp
#include "ItemAgenda.h"
#include <iostream>//entrada e saida

using namespace std;

//construtor
ItemAgenda::ItemAgenda(string descricao)
{
	if (!setDescricao(descricao))
	{
		this->descricao = " ";
	}
}

//construtor

ItemAgenda::ItemAgenda(DataHora& data, Paciente& paciente, string descricao)
{
	this->data = data;
	this->paciente = paciente;

	if (!setDescricao(descricao))
	{
		this->descricao = " ";
	}
}

//construtor
ItemAgenda::ItemAgenda(ItemAgenda& pagamentoPaciente)
{

	this->data = pagamentoPaciente.data;
	this->paciente = pagamentoPaciente.paciente;
	this->descricao = pagamentoPaciente.descricao;
}

//destrutor

ItemAgenda:: ~ItemAgenda() {}

//getters

string ItemAgenda::getDescricao()
{
	return descricao;
}

//setters

bool ItemAgenda::setDescricao(string descricao)
{

	if (descricao == "" || descricao == " ")//conferir se nome eh valido, validacao fraca mas ajuda um pouco
	{
		return false;
	}

	this->descricao = descricao;

	return true;
}


//sobrecarga do operador de atribuicao '=' 

ItemAgenda& ItemAgenda:: operator=(ItemAgenda& item)
{
	if (this == &item)
	{
		return *this;
	}

	this->data = item.data;
	this->paciente = item.paciente;
	this->descricao = item.descricao;

	return *this;
}

//sobrecarga do operador 'cout'
ostream& operator << (ostream& out, ItemAgenda& item)
{
	out << "Data: " << item.data
		<< "  Paciente: " << item.paciente.getNome() << " CPF:" << item.paciente.getCpf() << "  " << item.descricao;
	return out;
}
